setup.py

```diff
@@ -31,9 +33,27 @@ def is_requirement(line):
     """
     return line and not line.startswith(('-r', '#', '-e', 'git+', '-c'))
 
+
+def get_version(*file_paths):
+    """
+    Extract the version string from the file at the given relative path fragments.
+    """
+    filename = os.path.join(os.path.dirname(__file__), *file_paths)
+    with open(filename, encoding='utf-8') as opened_file:
+        version_file = opened_file.read()
+        version_match = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]",
+                                  version_file, re.M)
```

Comment:
I'm sorry to comment on this after it's been applied to repos already.  I would like to have a way to see these changes that will be applied to many repos before we start applying them broadly.
```suggestion
        version_match = re.search(r"(?m)^__version__ = ['\"]([^'\"]+)['\"]", version_file)
```
1. I find the `(?m)` syntax for flags makes it easier to understand how they apply to the pattern.
2. You want `+` instead of `*` since you don't want an empty version string.
3. We wrap to 120 columns, so no need to break this across lines.